internal/serving/testing/minimatch.go

```diff
@@ -0,0 +1,173 @@
+	}
+	return mm, closer, nil
+}
+
+func createOpenMatchServer(paramsList []*serving.ServerParams) (*MiniMatchServer, error) {
+	logger := log.WithFields(paramsList[0].BaseLogFields)
+
+	// Add a hook to the logger to log the filename & line number.
+	log.SetReportCaller(true)
+
+	cfg := viper.New()
+	cfg.Set("logging.level", "debug")
+	cfg.Set("logging.format", "text")
+	cfg.Set("logging.source", true)
+
+	promPort, err := NextPort()
+	if err != nil {
+		return nil, fmt.Errorf("cannot create port for prometheus, %s", err)
+	}
+
+	cfg.Set("metrics.port", promPort)
+	cfg.Set("metrics.endpoint", "/metrics")
+	cfg.Set("metrics.reportingPeriod", "5s")
+
+	// Configure open match logging defaults
+	logging.ConfigureLogging(cfg)
+
+	// Configure OpenCensus exporter to Prometheus
+	// metrics.ConfigureOpenCensusPrometheusExporter expects that every OpenCensus view you
+	// want to register is in an array, so append any views you want from other
+	// packages to a single array here.
+	ocServerViews := []*view.View{}
+	/* TODO: Views are conflicting, so not loading them.
+	for _, params := range paramsList {
+		ocServerViews = append(ocServerViews, params.CustomMeasureViews...)
+	}
+	*/
+	ocServerViews = append(ocServerViews, ocgrpc.DefaultServerViews...)      // gRPC OpenCensus views.
+	ocServerViews = append(ocServerViews, config.CfgVarCountView)            // config loader view.
+	ocServerViews = append(ocServerViews, redis.ObservabilityMetricViews...) // redis OpenCensus views.
+	logger.WithFields(log.Fields{"viewscount": len(ocServerViews)}).Info("Loaded OpenCensus views")
+	metrics.ConfigureOpenCensusPrometheusExporter(cfg, ocServerViews)
+
+	// Connect to redis
+	mredis, err := miniredis.Run()
+	if err != nil {
+		return nil, err
+	}
+
+	cfg.Set("redis.hostname", mredis.Host())
+	cfg.Set("redis.port", mredis.Port())
+	cfg.Set("redis.pool.maxIdle", 1000)
+	cfg.Set("redis.pool.idleTimeout", time.Second)
+	cfg.Set("redis.pool.maxActive", 1000)
+	cfg.Set("playerIndices", []string{
+		"char.cleric",
+		"char.knight",
+		"char.paladin",
+		"map.aleroth",
+		"map.oasis",
+		"mmr.rating",
+		"mode.battleroyale",
+		"mode.ctf",
+		"mode.demo",
+	})
+
+	pool, err := redishelpers.ConnectionPool(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	grpcPort, err := NextPort()
+	if err != nil {
+		return nil, fmt.Errorf("cannot provision a port for gRPC, %s", err)
+	}
+	for _, params := range paramsList {
+		cfg.Set(params.PortConfigName, grpcPort)
+	}
+
+	// Instantiate the gRPC server with the connections we've made
+	logger.Info("Attempting to start gRPC server")
+	grpcServer := serving.NewGrpcServer(grpcPort, logger)
+
+	mmServer := &MiniMatchServer{
+		OpenMatchServer: &serving.OpenMatchServer{
+			Config:     cfg,
+			GrpcServer: grpcServer,
+			Logger:     logger,
+			RedisPool:  pool,
+		},
+		mRedis: mredis,
+	}
+	for _, params := range paramsList {
+		for _, binding := range params.Bindings {
+			err := binding(mmServer.OpenMatchServer)
+			if err != nil {
```

Comment:
is there anything which needs to be stopped if an error occurs here?

Reply:
Added a fix for this and TODO for a cleaner solution in the next PR.